lib/SILOptimizer/Utils/Generics.cpp

```diff
@@ -2610,3 +2611,457 @@ SILFunction *swift::lookupPrespecializedSymbol(SILModule &M,
   return Specialization;
 }
 
+/// Create a new function name for the newly generated protocol constrained generic function.
+std::string ExistentialSpecializerTransform::createExistentialSpecializedFunctionName(){
+  SILModule &M = F->getModule();
+  int UniqueID = 0;
+  std::string MangledName;
+  do {
+    MangledName = Mangler.mangle(UniqueID);
+    ++UniqueID;
+  } while (M.hasFunction(MangledName));
+  return MangledName;
+}
+
+/// Recursively add composite types to the generic signature requirement.
+static void recursivelyAddCompositeTypes(ProtocolCompositionType *PType, 
```

Comment:
Not sure I quite follow you. Can you elaborate more or point me to existing code that does this?

Reply:
In this loop below:

```
/// Convert the protocol arguments of F to generic ones.
```

You add a requirement if the type is a protocol:

```
+      Requirement NewRequirement(RequirementKind::Conformance, NewGenericParam, DeclType->getDeclaredType());
 +      Requirements.push_back(NewRequirement);
```

Otherwise, you explode it recursively. But in fact you can create a `Requirement` even when the right hand side is a ProtocolCompositionType.